### PR TITLE
Highlight user journey path

### LINKED ISSUE #17: Highlight dependency path through graph for a given user journey

### PR DESCRIPTION
**Pull request by riverchen99**
closes #17

### CODE REVIEW

File: ujt/callbacks.py [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+            node_name_message_map,
+            client_name_message_map,
+        )
+    else:
+        if selected_row_ids != [None]:
+            user_journey_name = selected_row_ids[0][0]
+        else:
+            user_journey_name = None
+
+        cytoscape_graph_elements = transformers.apply_highlighted_edge_class_to_elements(
```
Comment (riverchen99) [inline ujt/callbacks.py:86]: These callbacks were previously in app.py, now refactored them to their own module.

Here's where the callback calls the helper function to apply the highlighting to relevant edges.

File: ujt/dash_app.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+import dash
+import dash_bootstrap_components as dbc
+import dash_cytoscape as cyto
+from flask_caching import Cache
+
+# Initialize Dash app and Flask-Cache
+cyto.load_extra_layouts()
+app = dash.Dash(__name__, external_stylesheets=[dbc.themes.BOOTSTRAP])
+cache = Cache()
```
Comment (riverchen99) [inline ujt/dash_app.py:15]: This module is loaded before we start the server, and exposes app and cache for other modules to use.

File: ujt/state.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
+from typing import Dict, Tuple, cast
+
+from graph_structures_pb2 import Client, Node
+
+from . import compute_status, generate_data, utils
+from .dash_app import cache
+
+
+def clear_cache():
```
Comment (riverchen99) [inline ujt/state.py:9]: Moved all the state modifying methods into one file so different parts of the code can access the underlying data through a unified interface.

File: ujt/transformers.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    return list(updated_edges_map.values()) + nodes_list
+
+
+def remove_highlighted_class_from_edges(edges_map):
+    for edge in edges_map.values():
+        edge["classes"] = ""
+    return edges_map
+
+
+def apply_highlighted_class_to_edges(edges_map, user_journey_name):
```
Comment (riverchen99) [inline ujt/transformers.py:40]: We use a BFS traversal to apply the a highlighted class to the edges so they stylesheet can color them differently.
Chose to use iteration over recursion here as it's easier to test. 

Note that here we traverse the actual protos (node_name_message_map). A future branch refactors this to only traverse the Cytoscape elements (to support virtual nodes)

File: ujt/transformers.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
+""" Module holding transformer functions.
+
+Transformer functions take a Dash-specific data structure or an intermediate data structure,
+make a change, and return the same type of data structure.
```
Comment (pwat2) [inline ujt/transformers.py:4]: what kind of changes can we expect to be made? 
is the invariant here the type of data structure being used or is there a difference between Dash and intermediate structures?

**Review by pwat2 — APPROVED** [review]

File: ujt/transformers.py (hunk 5 above)
Comment (riverchen99) [reply ujt/transformers.py:4]: Agreed, the docstring phrasing is confusing, I'll try to modify it.
 
For all the cases we currently have, transformer signatures are of the form: foo(cytoscape_elements, other_args) -> cytoscape_elements.
The changes currently being made include adding/removing nodes or edges, or modifying classes for styling.

I was trying to include the case where we have some generic type and a transformer bar(my_type, other_args) -> my_type in the docstring, but I don't have any code that falls into this category yet.

**Pull request merged**